initrd/directory.go

```diff
@@ -155,6 +157,39 @@ func (initrd *directory) Build(ctx context.Context) (string, error) {
 		return "", fmt.Errorf("could not walk output path: %w", err)
 	}
 
+	if initrd.opts.compress {
+		writer.Close()
+		f.Close()
+
+		fr, err := os.OpenFile(initrd.opts.output, os.O_RDONLY, 0o644)
+		if err != nil {
+			return "", fmt.Errorf("could not open initramfs file: %w", err)
+		}
```

Comment:
Instead of doing this you could just [Seek](https://pkg.go.dev/os#File.Seek) back to the beginning of the file with `f.Seek(os.SEEK_SET, 0)`.

Reply:
hmm, makes sense, I can do that yes. I was more worried about if the data is fully present there (hence the closes before)

Reply:
did, seems to work fine 👍